File: src/main/python/SparkConfig.py
import sys
from pyspark import SparkConf
from pyspark import SparkContext

if __name__ == "__main__":
  if len(sys.argv) < 2 or ( sys.argv[1] != "compileTime" and sys.argv[1] != "runTime" ):
    print >> sys.stderr, "Usage: SparkConfig <compileTime|runTime>"
    exit(-1)

  if sys.argv[1] == "compileTime":
    conf = SparkConf().setAppName("CT App")
    sc = SparkContext(conf=conf)
    print("Application Name: " + sc.appName)
    sc.stop()

  else:
    sc = SparkContext()
    print("Application Name: " + sc.appName)
    sc.stop()

File: src/main/scala/my/spark/examples/SparkConfig.scala
package my.spark.examples

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object SparkConfig {
  def main(args: Array[String]) {

    if (args.length < 1 || (args(0) != "compileTime" && args(0) != "runTime") ) {
      System.err.println("Usage: SparkConfig <compileTime|runTime>")
      System.exit(1)
    }

    if (args(0) == "compileTime") {
      val conf = new SparkConf().setAppName("CT App")
      val sc = new SparkContext(conf)
      println("Application Name: " + sc.appName)
      sc.stop()
    }
    
    else {
      val sc = new SparkContext()
      println("Application Name: " + sc.appName)
      sc.stop()
    }
  }
}
